desc/compute/_curve.py

```diff
@@ -156,6 +157,7 @@ def _phi_Curve(params, transforms, profiles, data, **kwargs):
     parameterization="desc.geometry.core.Curve",
 )
 def _Z_Curve(params, transforms, profiles, data, **kwargs):
+    errorif(kwargs.get("basis", "rpz").lower()[-1] != "z", NotImplementedError)
```

Comment:
It would be safer to check for `"rpz"` instead of just `"z"`, in case we add more bases in the future that might use z to mean something different (e.g., `"rtz"` the z means zeta instead of Z). Same for other files where you have this check for only z